record_linux/lib/record_linux.dart

```diff
@@ -251,90 +191,83 @@ class RecordLinux extends RecordPlatform {
-        continue;
-      }
-
-      if (line.startsWith(RegExp(r'^device #'))) {
-        // Extract previous device if second line was missing
-        extract();
-        deviceLine = line;
-      } else if (line.startsWith(RegExp(r'^\s*Default Format:'))) {
-        extract(secondLine: line);
+    if (currentDeviceId != null && currentDeviceName != null) {
+      if (!currentDeviceName.startsWith('Monitor of')) {
+        devices.add(InputDevice(id: currentDeviceId, label: currentDeviceName));
       }
     }
 
-    // Extract previous device if second line was missing
-    extract();
-
     return devices;
   }
 
-  InputDevice? _extractDevice(String firstLine, {String? secondLine}) {
-    final match = RegExp(r'(?:.*device #)(\d+): (\w.+)').firstMatch(firstLine);
-    if (match == null || match.groupCount != 2) return null;
+  @override
+  Future<List<InputDevice>> listInputDevices(String recorderId) async {
+    final outStreamCtrl = StreamController<List<int>>();
 
-    // ID
-    final id = match.group(1);
-    if (id == null) return null;
+    final out = <String>[];
+    outStreamCtrl.stream
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((chunk) {
+      out.add(chunk);
+    });
 
-    // Label
-    var label = match.group(2)!;
-    // Remove default from label
-    final index = label.indexOf(' - Default');
-    if (index != -1) {
-      label = label.substring(0, index);
+    try {
+      await _callPactl(['list', 'sources'], recorderId: recorderId, outStreamCtrl: outStreamCtrl);
+
+      return _parseInputDevices(out);
+    } finally {
+      outStreamCtrl.close();
     }
+  }
 
-    // int? channels;
-    // int? sampleRate;
-    // if (secondLine != null) {
-    //   final match = RegExp(
-    //     r'(?:.*Default Format: )(\d+) channel, (\d+) Hz',
-    //   ).firstMatch(secondLine);
-
-    //   if (match != null && match.groupCount == 2) {
-    //     // Number of channels
-    //     final channelsStr = match.group(1);
-    //     channels = channelsStr != null ? int.tryParse(channelsStr) : null;
-
-    //     // Sampling rate
-    //     final samplingStr = match.group(2);
-    //     sampleRate = samplingStr != null ? int.tryParse(samplingStr) : null;
-    //   }
-    // }
-
-    return InputDevice(
-      id: id,
-      label: label,
-      // channels: channels,
-      // sampleRate: sampleRate,
+  @override
+  Stream<RecordState> onStateChanged(String recorderId) {
+    _stateStreamCtrl ??= StreamController(
+      onCancel: () {
+        _stateStreamCtrl?.close();
+        _stateStreamCtrl = null;
+      },
     );
+
+    return _stateStreamCtrl!.stream;
+  }
+
+  List<String> _getEncoderSettings(AudioEncoder encoder, String path, int bitRate) {
+    switch (encoder) {
+      case AudioEncoder.aacLc:
+      case AudioEncoder.aacHe:
+        return ['-c:a', 'aac', '-b:a', '${bitRate/1000}k', path];
+      case AudioEncoder.wav:
+        return ['-c:a', 'pcm_s16le', path];
```

Comment:
I guess this may lead to issues here. As far as I understand if file extension is wav this will provide WAVE format?

Reply:
I have improved this in a way that flac and wav are directly done by `parecord` which can handle them. The others work just fine with ffmpeg (I tried wav, flac, and aac). 

Not sure if you see any improvements room?